temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowStoreImpl.java

```diff
@@ -556,7 +526,17 @@ private WorkflowExecutionInfo constructWorkflowExecutionInfo(
 
   @Override
   public void close() {
-    isShuttingDown = true;
-    timerService.shutdown();
+    lock.lock();
+    try {
+      if (isShuttingDown) {
+        return;
+      }
+      isShuttingDown = true;
+      activityTaskQueues.values().forEach(TaskQueue::close);
```

Comment:
I'm not sure what problem does the closing of TaskQueue solves (and the existence of `TaskQueue`).
It looks like we close them when we close the whole TestService only. The first thing we do when we close 


TestService actually stops GRPC's Server and corresponding Client. This should trigger interruption of the thread waiting on the long polls and prevention of anything new to be added to the queues too. 

Is this broken somehow? Was it maybe broken by this newly added `withContextInterruption` that swallows interrupted flag that is installed not by `interrupter`?
Let's maybe fix `withContextInterruption` and reevaluate the need for TaskQueue after that?

Reply:
Well this may be rather more than is needed but the thinking here is that there are 3 ways to stop a grpc-java server
1. `shutdown()` which does _not_ interrupt in-flight RPCs
2. `shutdownNow()` which _does_ interrupt in-flight RPCs
3. Just exit the process

Right now we appear to do number three when running out-of-proc. But I take your point - it would be preferable to have appropriate startup/shutdown code.